### PR TITLE
Added Bead Sort in C

### LINKED ISSUE #2072: Bead Sort
## 🚀 Feature
[DWOC Contribution]
Bead Sort in C++

(A clear and concise description of what the feature is.)

### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#reporting-new-issues)?

Yes

## Motivation

Bead Sort (also known as Gravity Sort) was inspired from natural phenomenons and was designed keeping in mind objects(or beads) falling under the influence of gravity.

### PR DESCRIPTION
**Pull request by m-code12**
Fixes: #2072

### CODE REVIEW

**Comment by m-code12** [comment]
@HarshCasper @atarax665 Could you review this pr?

File: C/sort/beadSort.c [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+
+	for (k = 0; k < max; k++) {
+		sum=0;
+		for (i = 0; i < n; i++) {
+			sum = sum + BEAD(i, k);
+			BEAD(i, k) = 0;
+		}
+		for (i = n - sum; i < n; i++)
+		    BEAD(i, k) = 1;
+	}
```
Comment (HarshCasper) [inline C/sort/beadSort.c]: Push the core code logic into a separate function and use the main() function for driver code

**Review by HarshCasper — CHANGES_REQUESTED** [review]
Make the README changes as well. Fit to merge 👍

**Comment by m-code12** [comment]
@HarshCasper Made the changes

File: C/sort/beadSort.c [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+		arr[i] = k;
+	}
+
+	//deallocate the memory allocated to beads
+	free(beads);
+}
+
+int main(void) {
+
+	int n,i,k;
```
Comment (atarax665) [inline C/sort/beadSort.c]:
```suggestion
	int n, i, k;
```

File: C/sort/beadSort.c [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+	printf("Enter the number of elements in the array : ");
+	scanf("%d",&n);
+
+	int nums[n];
+	printf("Enter the elements of the array : ");
+	for(i=0;i<n;i++){
```
Comment (atarax665) [inline C/sort/beadSort.c]:
```suggestion
	for(i = 0; i < n; i++){
```

File: C/sort/beadSort.c [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+	    scanf("%d",&nums[i]);
+	}
+
+    bead_sort(nums, n);
```
Comment (atarax665) [inline C/sort/beadSort.c]: Inconsistent indentation.

**Review by atarax665 — CHANGES_REQUESTED** [review]
Just some minor fixes and then it would be good to merge.

**Comment by m-code12** [comment]
@atarax665 Made all the changes

**Review by HarshCasper — APPROVED** [review]

**Pull request merged**